models/RwdImage.php

```diff
@@ -464,6 +478,108 @@ public function get_set_sources() {
+				'rwd_height' => $height,
+				'crop'       => $crop,
+			);
+			// save to cache.
+			$this->set_attachment_metadata( $attach_id, $meta_data );
+			// update metadata.
+			wp_update_attachment_metadata( $attach_id, $meta_data );
+
+			return $meta_data;
+		}
+		// set RWD size config
+		if ( ! isset( $meta_data['sizes'][ $key ] ) ) {
+			$meta_data['sizes'][ $key ] = array(
+				'rwd_width'  => $width,
+				'rwd_height' => $height,
+				'crop'       => $crop,
+			);
+		} elseif ( ! isset( $meta_data['sizes'][ $key ]['rwd_width'] ) ) {
+			$meta_data['sizes'][ $key ] = array(
+				'rwd_width'  => $width,
+				'rwd_height' => $height,
+				'crop'       => $crop,
+			);
+		}
+		$upload_dir    = wp_get_upload_dir();
+		$image_baseurl = $upload_dir['basedir'] . DIRECTORY_SEPARATOR . image_get_intermediate_size( $attach_id, $key )['path'];
+		if ( $meta_data['sizes'][ $key ]['rwd_width'] !== $width || $meta_data['sizes'][ $key ]['rwd_height'] !== $height
+		     || $meta_data['sizes'][ $key ]['crop'] !== $crop || ! file_exists( $image_baseurl )
+		) {
+			// Get WP Image Editor Instance
+			$image_path   = get_attached_file( $attach_id );
+			$image_editor = wp_get_image_editor( $image_path );
+			if ( ! is_wp_error( $image_editor ) ) {
+				// Create new image
+				$auto_height = ( $height == 19998 || $height == 9999 ? null : $height );
+				// WP Image Editor resize
+				$image_editor->resize( $width, $auto_height, $crop );
+				// Generate filename
+				$resize_filename = basename( $image_editor->generate_filename() );
+				$resize_sizes    = $image_editor->get_size();
+				if ( $meta_data['width'] >= $resize_sizes['width'] && $meta_data['height'] > $resize_sizes['height'] ) {
+					// WP Image Editor save image
+					$image_editor->save();
+					$meta_data['sizes'][ $key ] = array(
+						'width'      => $resize_sizes['width'],
+						'height'     => $resize_sizes['height'],
+						'rwd_width'  => $width,
+						'rwd_height' => $height,
+						'crop'       => $crop,
+						'file'       => $resize_filename,
+						'mime-type'  => get_post_mime_type( $attach_id ),
+					);
+					// save to cache.
+					$this->set_attachment_metadata( $attach_id, $meta_data );
+					// update metadata.
+					wp_update_attachment_metadata( $attach_id, $meta_data );
+				} else {
+					// use max image size for the bigger sizes.
+					if ( ! strpos( $key, '@' ) ) {
+						$meta_data['sizes'][ $key ] = array(
+							'width'      => $meta_data['width'],
+							'height'     => $meta_data['height'],
+							'file'       => basename( $meta_data['file'] ),
+							'rwd_width'  => $width,
+							'rwd_height' => $height,
+							'crop'       => $crop,
+							'mime-type'  => get_post_mime_type( $attach_id ),
+						);
+					} else {
+						unset( $meta_data['sizes'][ $key ] );
+					}
+					// save to cache.
+					$this->set_attachment_metadata( $attach_id, $meta_data );
+					// update metadata.
+					wp_update_attachment_metadata( $attach_id, $meta_data );
```

Comment:
these 2 lines can be placed after if/else. - in this way you will call it in 1 place.